addons/base_iban/models/base_iban.py

```diff
@@ -0,0 +1,183 @@
+        if vals.get('state') == 'iban' and vals.get('acc_number'):
+            vals['acc_number'] = _format_iban(vals['acc_number'])
+            vals['acc_number'] = _pretty_iban(vals['acc_number'])
+        return super(ResPartnerBank, self).create(vals)
+
+    @api.multi
+    def write(self, vals):
+        # overwrite to format the iban number correctly
+        if vals.get('state') == 'iban' and vals.get('acc_number'):
+            vals['acc_number'] = _format_iban(vals['acc_number'])
+            vals['acc_number'] = _pretty_iban(vals['acc_number'])
+        return super(ResPartnerBank, self).write(vals)
+
+    def is_iban_valid(self, iban):
+        """ Check if IBAN is valid or not
+            @param iban: IBAN as string
+            @return: True if IBAN is valid, False otherwise
+        """
+        if not iban:
+            return False
+        iban = _format_iban(iban).lower()
+        if iban[:2] in _ref_iban and len(iban) != len(_format_iban(_ref_iban[iban[:2]])):
+            return False
+        #the four first digits have to be shifted to the end
+        iban = iban[4:] + iban[:4]
+        #letters have to be transformed into numbers (a = 10, b = 11, ...)
+        iban2 = ""
+        for char in iban:
+            if char.isalpha():
+                iban2 += str(ord(char) - 87)
+            else:
+                iban2 += char
+        #iban is correct if modulo 97 == 1
+        return int(iban2) % 97 == 1
+
+    @api.one
+    @api.constrains('acc_number', 'state')
+    def check_iban(self):
+        '''
+        Check the IBAN number
+        '''
+        def default_iban_check(iban_cn):
+            return iban_cn and iban_cn[0] in string.ascii_lowercase and iban_cn[1] in string.ascii_lowercase
+        for bank_acc in self:
+            if bank_acc.state != 'iban':
+                continue
+            if not self.is_iban_valid(bank_acc.acc_number):
+                iban_country = self.acc_number and self.acc_number[:2].lower()
+                if default_iban_check(iban_country):
+                    if iban_country in _ref_iban:
+                        raise UserError(_('The IBAN does not seem to be correct. You should have entered something like this %s' \
+                                         '\n Where B = National bank code, S = Branch code,'
+                                          ' C = Account No, K = Check digit' % _ref_iban[iban_country]))
+                    raise UserError(_('This IBAN does not pass the validation check, please verify it'))
+                raise UserError(_('The IBAN is invalid, it should begin with the country code'))
+
+    @api.one
+    @api.constrains('bank_bic')
+    def _check_bank(self):
+        for partner_bank in self:
+            if partner_bank.state == 'iban' and not partner_bank.bank_bic:
+                raise UserError(_('Please define BIC/Swift code on bank for bank type IBAN Account to make valid payments'))
+
+    @api.multi
+    def get_bban_from_iban(self):
```

Comment:
@tde-banana-odoo    @rim-odoo 
I was migrating `base_iban` module and in that i found that is method is not used anywhere. So shall i remove it..??